src/components/checkbox/checkbox.tsx

```diff
@@ -0,0 +1,48 @@
+import React = require('react');
+
+interface CheckBoxProps {
+    text?: string;
+    value?: boolean;
+    boxIcon?: string;
+    tickIcon?: string;
+    onChange?: (value: boolean ) => any;
+
+}
+
+export class CheckBox extends React.Component<CheckBoxProps, {}> {
```

Comment:
I don't have a concrete answer to give you apart from conforming to a single pattern... maybe @tobich will have a better answer